pallets/xyk/src/lib.rs

```diff
@@ -433,9 +433,11 @@ impl<T: Trait> Module<T> {
         let output_reserve_saturated: U256 = output_reserve.into();
         let sell_amount_saturated: U256 = sell_amount.into();
 
-        let input_amount_with_fee: U256 = sell_amount_saturated * 997;
-        let numerator: U256 = input_amount_with_fee * output_reserve_saturated;
-        let denominator: U256 = input_reserve_saturated * 1000 + input_amount_with_fee;
+        let input_amount_with_fee: U256 = sell_amount_saturated.saturating_mul(997.into());
+        let numerator: U256 = input_amount_with_fee.saturating_mul(output_reserve_saturated);
+        let denominator: U256 = input_reserve_saturated
+            .saturating_mul(1000.into())
```

Comment:
What is the difference between `saturating` and `checked` operations? Why `saturating` is chosen?